lib/stdlib/test/supervisor_SUITE.erl

```diff
@@ -3727,6 +3729,70 @@ already_started_outside_supervisor(_Config) ->
     ok = check_exit([SupPid]),
     ok.
 
+%% Test which_children/1 and which_child/2.
+which_children(Config) when is_list(Config) ->
+    {ok, SupPid} = start_link({ok, {#{}, []}}),
+
+    [] = supervisor:which_children(SupPid),
```

Comment:
is this the 99% same test code in 2 tests? maybe consider having a helper function for easier maintenance in future - so that redundant code is reduced.

Reply:
tl;dr, yes it is 😅

I can do it as you suggest and will readily yield to any such decision.

However, personally, I would argue that those tests test behaviors which are merely _very similar_, but not the _same_, and that this fact justifies two distinct fixed tests, even if they are also very similar to each other.

But you decide 😉

Reply:
I think it is a tradeoff. Sometimes it could be harder to understand the test case if you sort or loose the context.
It could of course be compensated with good variable naming's. I am not sure what I would think in this case without an example of how the helper should look.

Reply:
I was tired when doing review. My bad. After re-reading the code, I don't think it is worth changing.
Sorry for confusion. Pls, leave it as it is.

Reply:
@u3s np, will do :hugs:

Reply:
That said, I just augmented the `which_children_simple_one_for_one` test a little, to check that giving a non-pid to a `simple_one_for_one` supervisor results in `{error, simple_one_for_one}`.